### ISSUE TITLE
Client docs: Cyberduck vs. Lighttpd (Chunked encoding)

### ISSUE THREAD

**Issue by RealRancor** [fruux/sabre.io#30]
Hi,

currently the http://sabre.io/dav/clients/finder contains a note that lighttpd (and older NginX versions) are unsupported because of the used Chunked encoding.

It seems that recent Cyberduck versions are also using Chunked encoding:

https://trac.cyberduck.io/ticket/5603
https://trac.cyberduck.io/ticket/6391

so the note of Finder is probably also valid for Cyberduck?

**Comment by evert** [comment]
Hi!

Both of the tickets you quoted are not about the webdav protocol, and more importantly: those chunked headers are in the context of the HTTP *responses* and not the *requests*. The issue you quote only has to do with clients making requests.

**Comment by RealRancor** [comment]
Mhhh, had a chat with some one at the #owncloud IRC channel running lightttpd on ownCloud.

He had problems uploading files larger than 2MB. In Finder he got an:

> The Finder can’t complete the operation because some data in “MyFile.mp4” can’t be read or written.(Error code -36)

message. Cyberduck has also failed with files larger than 2MB (no error message was posted).

On the other hand the "Documents" app on iPhone as well as the Interarchy client worked for him uploading larger than this 2MB limit.

**Comment by evert** [comment]
A sane thing to do, also for the owncloud people is to simply not at all support lighttp. Afaik it's a dead project, and still has plenty of bugs. There's better alternatives.

**Comment by RealRancor** [comment]
Hi,

ok good to know. Thanks for the info. Should we close this one?

**Comment by Hywan** [comment]
@RealRancor Yup :-).

**Comment by evert** [comment]
Hi @gstrauss ,

My experience in the past with the lighttpd project was that bugs were left open for years until they were addressed, hence my negativity. I did opening issues and even submitted patches.It's great to hear lighttpd has made some progress though. Are arbitrary HTTP methods supported yet, or is there still a hardcoded list?

**Comment by gstrauss** [comment]
Unfortunately, lighttpd did appear to stagnate for many years as bugs were left open for years.  That changed in a big way in 2016.  Hundreds of bugs have been fixed and many feature requests have been addressed.  While there is still more to do, work is ongoing.

Regarding "arbitrary" request methods, if lighttpd does not recognize a method defined in RFCs, please consider submitting a feature request and we can extend the list of recognized methods in src/keyvalue.c.  It would be a bit more work to support arbitrary methods, but I am open to being convinced of the value of such a feature, and submitted patches will be reviewed.  Cheers.

**Comment by novat0** [comment]
i want to apport a 50c:

lighttp developers have family and personal life, event make reclaims to lighttp project, just colaborate with! what its happened here!? its more easy that owncloud inc-close people reclaim event make goods apport?

**Comment by gstrauss** [comment]
@novat0 thanks for the vote of confidence (I think).  Your message seems to be somewhat lost in translation.  In any case, I do not think your attack is warranted.  Please do not attack others.

**Comment by evert** [comment]
I'm also having trouble interpreting @novat0, but I want to stress that before I started telling people to stop using Lighttpd, I did reach out and try to improve the situation. It was only after not getting responses that I felt that it was in the best interest of end-users to not use lighttpd. This was never a judgement of the people behind lighttpd, but simply coming from my own experiences that lighttpd neither supported the features sabre/dav needed to work and literally waiting years to get responses to my reports.

That being said, it sounds that things have turned around. I will definitely stop spreading this and hopefully some people can tell me their experiences with sabre/dav and lighttpd are now positive. For now I will continue to recommend Nginx and Apache when asked, simply because it has a larger user-base and even in those servers it tends to be hard for people to configure stuff. But I'll no longer discourage people from using lighttpd if they prefer it.
